### PR TITLE
feat: support azure openai

### PR DESCRIPTION
**Pull request by wanghanzhen**
### Description

support azure openai.

### CODE REVIEW

File: src/features/ai/completions/openai.ts [hunk 1]
```diff
@@ -56,25 +55,62 @@ export async function OpenAIChatCompletion(request: FastifyRequest, reply: Fasti
...
+
+  let stream: Stream<ChatCompletionChunk> | EventStream<ChatCompletions>
+
+  if (openaiConfig?.isAzure) {
+    const azureOpenai = new OpenAIClient(
+      openaiConfig!.baseUrl!,
+      new AzureKeyCredential(openaiConfig!.apiKey!),
+    )
+    stream = await azureOpenai.streamChatCompletions(
+      openaiConfig.azureDeploymentId || body.model,
```
Comment (wibus-wee) [inline src/features/ai/completions/openai.ts:67]: Would like to ask why `openaiConfig.azureDeploymentId` is the highest priority here?

**Review by wibus-wee — CHANGES_REQUESTED** [review]
Additionally, for a better user experience, it may be necessary to add new models in `src/routes/ai/constants.ts` to support Azure's models, or to modify `src/routes/ai/model.ts` so that it is visible in Raycast Settings.
CleanShot 2024-02-08 at 16 52 25@2x

**Comment by wanghanzhen** [comment]
> Additionally, for a better user experience, it may be necessary to add new models in `src/routes/ai/constants.ts` to support Azure's models, or to modify `src/routes/ai/model.ts` so that it is visible in Raycast Settings. CleanShot 2024-02-08 at 16 52 25@2x

My thought  is no one will use both OpenAI and Azure OpenAI at the same time, so I add a bool field `is_azure` in config.toml to toggle it simply and intuitively.

**Comment by wibus-wee** [comment]
> > Additionally, for a better user experience, it may be necessary to add new models in `src/routes/ai/constants.ts` to support Azure's models, or to modify `src/routes/ai/model.ts` so that it is visible in Raycast Settings. CleanShot 2024-02-08 at 16 52 25@2x
> 
> 
> 
> My thought  is no one will use both OpenAI and Azure OpenAI at the same time, so I add a bool field `is_azure` in config.toml to toggle it simply and intuitively.

So is the model of azure openai actually the same as that of openai?

I haven't used azure openai api, sorry.

**Comment by wanghanzhen** [comment]
> maybe you can make a little notes in the readme🥰🥰

done and changed the `providerName` back.

**Review by wibus-wee — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**